larky/src/main/java/com/verygood/security/larky/modules/types/results/Result.java

```diff
@@ -0,0 +1,172 @@
+package com.verygood.security.larky.modules.types.results;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import net.starlark.java.annot.Param;
+import net.starlark.java.annot.StarlarkMethod;
+import net.starlark.java.eval.Dict;
+import net.starlark.java.eval.EvalException;
+import net.starlark.java.eval.Printer;
+import net.starlark.java.eval.Starlark;
+import net.starlark.java.eval.StarlarkFunction;
+import net.starlark.java.eval.StarlarkThread;
+import net.starlark.java.eval.StarlarkValue;
+import net.starlark.java.eval.Tuple;
+
+public class Result implements StarlarkValue {
+
+  private Object value;
+  private EvalException error;
+
+  Result(Object value, EvalException error) {
+    this.value = value;
+    this.error = error;
+  }
+
+  @StarlarkMethod(name = "failure", parameters = {@Param(name = "error")})
+  public static Result failure(Object error) {
+    Objects.requireNonNull(error);
+    if (EvalException.class.isAssignableFrom(error.getClass())) {
+      return new Result(null, (EvalException) error);
+    }
+    return new Result(null, new EvalException(String.valueOf(error)));
+  }
+
+  @StarlarkMethod(name = "success", parameters = {@Param(name = "value")})
+  public static Result success(Object value) {
+    Objects.requireNonNull(value);
+    return new Result(value, null);
+  }
+
+  @StarlarkMethod(name = "of", parameters = {@Param(name = "o")})
+  public static Result of(Object o) {
+    if (o instanceof Exception) {
+      return failure(o);
+    }
+    return success(o);
+  }
+
+  @StarlarkMethod(name = "either")
+  public Object getEither() {
+    return value != null ? value : error;
+  }
+
+  @StarlarkMethod(name = "value")
+  public Object getValue() {
+    return value;
+  }
+
+  @StarlarkMethod(name = "error")
+  public EvalException getError() {
+    return error;
```

Comment:
As we provide accessors to underlying `Error` primitive, I think we should make an `Error` object as a legitimate first-class-citizen object of Larky.
 Which will allow doing sth like:
```
result  = doSomeFancyBusinessLogic()
err = result.getError()
if err.status() == 404:
    ...
elif err.status() == 401:
    ...
else:
    return Error(500)
```
I've intentionally used HTTP status codes, as they cover almost all use-cases (even IoT 418), familiar to everyone, and easy to interop.

Reply:
https://github.com/verygoodsecurity/starlarky/pull/121/commits/850b5cac5236a5c73cb3659b453020c72218d23a